Pasing from SAS to R second program

```diff
@@ -0,0 +1,102 @@
+Moving Second Program from SAS to R
+
+This are the notes that apper as I move forward. I have the code in a separate document. 
+
+
+line 28 of SAS, I don't know why are we deleting the data that is NOT 1. 
+
+
+To beging the move to R, I'm going to import the following 3 datasets from SAS
+possiblenclass
+citations
+matching
+
+Line 217 - 234, just before the firts iteration I beging to export this dataset to csv files in order to take them to R. 
+
+I added the following code to the second SAS program. 
+
+/*Export the date to csv files in order to continue the next step in R*/
+/* I rename the datasets SASname to distinguish from the ones in SAS*/
+
+proc export data= work.matching dbms=csv
+	outfile= "\\hd.ad.syr.edu\01\e88665\Documents\Desktop\Backup SAS\from SAS to R\SASmatching.csv"
+	replace;
+run;
+
+proc export data= work.citations dbms=csv
+	outfile= "\\hd.ad.syr.edu\01\e88665\Documents\Desktop\Backup SAS\from SAS to R\SAScitations.csv"
+	replace;
+  run;
+
+proc export data= work.possiblenclass dbms=csv
+outfile= "\\hd.ad.syr.edu\01\e88665\Documents\Desktop\Backup SAS\from SAS to R\SASpossiblenclass.csv"
+	replace;
+run;
+
+
+
+In RStudio
+
+#I first beging by opening the working directory in which I'm going to work
+
+> setwd(file.path("//hd.ad.syr.edu","01","e88665","Documents","Desktop","Backup SAS","from SAS to R"))
+
+#Read the data from database matching
+> matching<- file.path("SASmatching.csv")
+> matching<-read.csv("SASmatching.csv")
+
+#Read the data from database citations
+> citations<- file.path("SAScitations.csv")
+> citations<- read.csv("SAScitations.csv")
+
+#Read the data from database possiblenclass
+> possiblenclass <- file.path("SASpossiblenclass.csv")
+> possiblenclass <- read.csv("SASpossiblenclass.csv")
+> View(possiblenclass)
+
+
+# Firts Iteration 
+
+#I replace the control numbers in the variable x_control with the uniform distribution
+> possiblenclass$x_control<-runif(nrow(possiblenclass))
+> View(possiblenclass$x_control)
+
+#After I assign random values I need to sort the dataset possiblenclass
+> possiblenclass[order(possiblenclass$nclass,possiblenclass$x_control),]
+
+# Next step is to create the dataset matching1 
+> matching1<-matching[c("cited","patent")]
+> View(matching1)
+> matching1$binvar<- 1
+
+# Line 233 call the dataset citation and sort it. 
+> citations[order(citations$cited,citations$patent),]
+
+#Line 237
+I'm confuse with this part, the code in SAS is the following 
```

Comment:
I'm going to need to take some time with this to understand it when I'm better rested. I should be able to do this either on Sunday or Monday. We can plan to meet to talk it over and get started on the gradualism project on Monday afternoon if that works for you.

In the meantime, Jake had suggested that we would get the best speed by using "Data Tables instead of standard R data frames." I asked a friend who is very good at R programming, and he agreed.

So why don't you take some time to learn what you can about the data.table package? https://cran.r-project.org/web/packages/data.table/vignettes/datatable-intro.html

Also, you can gather your questions about the gradualism project for Monday.